uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -0,0 +1,492 @@
+define([
+  'leave-absences/shared/modules/controllers',
+  'common/lodash',
+  'common/moment',
+  'common/services/api/option-group',
+  'common/services/hr-settings',
+], function (components, _, moment) {
+  'use strict';
+
+  components.controller('LeaveRequestPopupCtrl', [
+    '$log', '$q', '$uibModalInstance', 'AbsencePeriod', 'AbsenceType', 'Entitlement',
+    'Calendar', 'LeaveRequestInstance', 'LeaveRequest', 'api.optionGroup', 'baseData',
+    'PublicHoliday', 'HR_settings', '$rootScope',
+    function ($log, $q, $modalInstance, AbsencePeriod, AbsenceType, Entitlement,
+      Calendar, LeaveRequestInstance, LeaveRequest, OptionGroup, baseData, PublicHoliday,
+      HR_settings, $rootScope) {
+      $log.debug('LeaveRequestPopupCtrl');
+
+      var $ctrl = this,
+        serverDateFormat = 'YYYY-MM-DD';
+
+      $ctrl.absenceTypes = [];
+      $ctrl.calendar = {};
+      $ctrl.leaveRequestDayTypes = [];
+      $ctrl.balance = {
+        opening: 0,
+        change: {
+          amount: 0,
+          breakdown: []
+        },
+        closing: 0
+      };
+      $ctrl.period = {};
+      $ctrl.pagination = {
+        totalItems: $ctrl.balance.change.breakdown.length,
+        filteredbreakdown: $ctrl.balance.change.breakdown,
+        currentPage: 1,
+        numPerPage: 5,
+        pageChanged: function () {
+          //filter breakdowns
+          var begin = ((this.currentPage - 1) * this.numPerPage),
+            end = begin + this.numPerPage;
+
+          this.filteredbreakdown = $ctrl.balance.change.breakdown.slice(begin, end);
+        }
+      };
+      $ctrl.uiOptions = {
+        showDatePickerFrom: false,
+        showDatePickerTo: false,
+        isChangeExpanded: false,
+        datePickerOptions: {
+          startingDay: 1,
+          showWeeks: false
+        },
+        isAdmin: false,
+        userDateFormat: HR_settings.DATE_FORMAT,
+        showBalance: false,
+        multipleDays: true
+      };
+      $ctrl.error = undefined;
+      $ctrl.loading = {};
+
+      // Create an empty leave request
+      $ctrl.leaveRequest = LeaveRequestInstance.init({
+        contact_id: baseData.contactId //resolved from directive
+      }, false);
+
+      (function () {
+        $ctrl.loading.absenceTypes = true;
+        AbsencePeriod.current().then(function (apInstance) {
+            $ctrl.period = apInstance;
+          })
+          .then(function () {
+            return initAbsenceTypesAndEntitlements().then(function () {
+              $ctrl.loading.absenceTypes = false;
+            });
+          })
+          .then(function () {
+            return initDayTypesAndStatus()
```

Comment:
Is it important to `return` this? It doesn't seem you are doing anything with it?